### DISCUSSION TITLE
How to set the interface type via nbshell/custom scripts?

### DISCUSSION THREAD

**Discussion by zshorvat** [netbox-community/netbox#8506] [Q&A] [answered]
Python 3.9.2 | Django 3.2.11 | NetBox 3.1.6

Good day everyone,

I'm trying to change the interface type on a device via custom scripts or netbox but apparently whatever value I'm assigning, it does not seem to be taking effect...

I might be doing something stupid, but I cannot figure out what I should be using to make the change.

```
>>> Interface.objects.get(name="ONB1",device__name="MG-P-ESXED21").type="10gbase-t"
>>> print(Interface.objects.get(name="ONB1",device__name="MG-P-ESXED21").type)
1000base-t
>>> Interface.objects.get(name="ONB1",device__name="MG-P-ESXED21").type="10GBASE-T (10GE)"
>>> print(Interface.objects.get(name="ONB1",device__name="MG-P-ESXED21").type)
1000base-t
>>> Interface.objects.get(name="ONB1",device__name="MG-P-ESXED21").type="10GBASE-T(10GE)"
>>> print(Interface.objects.get(name="ONB1",device__name="MG-P-ESXED21").type)
1000base-t
>>> Interface.objects.get(name="ONB1",device__name="MG-P-ESXED21").type="10gbase-t (10ge)"
>>> print(Interface.objects.get(name="ONB1",device__name="MG-P-ESXED21").type)
1000base-t
```

**Comment by kkthxbye-code** [comment, marked as answer]
You have to call .save() for the object to be persisted to the database.

Something like this (untested):

```
interface = Interface.objects.get(name="ONB1",device__name="MG-P-ESXED21")
interface.type = "10gbase-t"
interface.full_clean()
interface.save()
```

**Reply by zshorvat** [reply]
Thanks a lot!

I actually tested with saving, but your suggestion made me figure what I did wrong:

```
>>> print(Interface.objects.get(name="ONB1",device__name="MG-P-ESXED21").type)
1000base-t
>>> Interface.objects.get(name="ONB1",device__name="MG-P-ESXED21").type="10gbase-t"
>>> print(Interface.objects.get(name="ONB1",device__name="MG-P-ESXED21").type)
1000base-t
>>> Interface.objects.get(name="ONB1",device__name="MG-P-ESXED21").save()
>>> print(Interface.objects.get(name="ONB1",device__name="MG-P-ESXED21").type)
1000base-t
```

I didn't create a new variable for the interface object, and tried modifying the type and saving directly from the Interface.objec.get() result which didn't work and that took me to into a rabbit-hole :)

Long story short: laziness never pays off, make sure to store them objects in their variables when doing any ops on them.

Thanks again!
